### ISSUE TITLE
Bring our release practices in line with rest of JupyterHub org

### ISSUE THREAD

**Issue by yuvipanda** [jupyterhub/repo2docker#1280]
JupyterHub now has a very well thought out release system (https://github.com/jupyterhub/jupyterhub/blob/main/RELEASE.md). repo2docker has *some* documentation, but it's a bit out of date. We should update our docs to match rest of JupyterHub.

**Comment by consideRatio** [comment]
I looked at this to figure out details on the action points:

- [x] release automation in place
- [x] release automation rely on versioneer instead of tbump, its ok I think
- [x] changelog maintained (in markdown, using github-activity)
- [ ] release docs doesn't mention the changelog (copy/paste/adapt [from systemdspawner/RELEASE.md](https://github.com/jupyterhub/systemdspawner/blob/main/RELEASE.md) pending https://github.com/jupyterhub/jupyterhub-python-repo-template/pull/21)
- [ ] release docs doesn't mention the [conda-forge feedstock](https://github.com/conda-forge/jupyter-repo2docker-feedstock) (copy/paste/adapt [from jupyterhub-python-repo-template/RELEASE.md](https://github.com/jupyterhub/jupyterhub-python-repo-template/blob/main/RELEASE.md))
- [ ] release docs doesn't mention anything else like refreezing environments - should we do this before release and if so, linking to what docs on doing that?
- [ ] readme doesn't include badges for pypi, conda-forge, code coverage, issue tracking, help forum (copy/paste/adapt from [jupyterhub-python-repo-template readme](https://github.com/jupyterhub/jupyterhub-python-repo-template/blob/main/README.md))

**Comment by manics** [comment]
I don't think refreezing environments should be part of the release process:
- it increases the workload on the person managing the release
- ideally we also want to test the refrozen environments by deploying to mybinder.org

I think it's fine to mention it as something to consider when planning a release though.

**Comment by consideRatio** [comment]
> ideally we also want to test the refrozen environments by deploying to mybinder.org

Are we using repo2docker main there?

**Comment by manics** [comment]
Yes, e.g. https://github.com/jupyterhub/mybinder.org-deploy/pull/2643

That's why we need versioneer (or some other auto versioning tool)- so we can have the git sha as part of the version for repo2docker installed from main, and for the container image built from main.

**Comment by minrk** [comment]
Agree that freezing should not be part of making a release, but is a good thing to highlight _early_ in preparation for a release. Generally releases should be relatively long after freezes.
